### PR TITLE
New: Refresh classification

### PR DESCRIPTION
**Pull request by DanDeMicco**
Adds a refresh function which will refetch the file info. Changes to sidebar loading to appear correct width depending on ```isCollapsed```

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #428

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/BusyIndicator.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * @flow
+ * @file Busy indicator for cards
+ * @author Box
+ */
+
+import * as React from 'react';
+import LoadingIndicator from 'box-react-ui/lib/components/loading-indicator';
+import './BusyIndicator.scss';
+
+const BusyIndicator = () => (
```
Comment (priyajeet) [inline src/components/ContentSidebar/BusyIndicator.js]: kinda need this one in metadata too.
I will port it over to BRUI.

File: src/components/ContentSidebar/ContentSidebar.js [hunk 2]
```diff
@@ -628,7 +645,13 @@ class ContentSidebar extends PureComponent<Props, State> {
      */
     fetchFile(id: string, forceFetch: boolean = false): void {
         if (SidebarUtils.canHaveSidebar(this.props)) {
-            this.api.getFileAPI().file(id, this.fetchFileSuccessCallback, this.errorCallback, forceFetch, true);
+            this.api
+                .getFileAPI()
+                .file(id, this.fetchFileSuccessCallback, this.fetchFileErrorCallback, forceFetch, true);
+
+            this.setState({
+                isFileLoading: true
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: Should this be a generic? isBusy or isLoading?

File: src/components/_variables.scss [hunk 3]
```diff
@@ -29,6 +29,9 @@ $lighter-light-blue: rgba($light-blue, .4) !default;
 
 $font: Lato, 'Helvetica Neue', Helvetica, Arial, sans-serif;
 
+$sidebarTabsWidth: 60px;
```
Comment (priyajeet) [inline src/components/_variables.scss]: Move these local to where they are used, unless referenced more than 1 place?

File: src/components/ContentSidebar/ContentSidebar.js [hunk 4]
```diff
@@ -1450,14 +1501,15 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+            currentUserError,
+            isFileLoading
         }: State = this.state;
 
         const styleClassName = classNames(
             'be bcs',
             {
                 [`bcs-${view}`]: !!view,
-                'bcs-is-open': !!view
+                'bcs-is-open': !!view || !isCollapsed
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: make sure to test responsive mode, shrinking viewport

File: src/components/ContentSidebar/ContentSidebar.js (hunk 2 above)
Comment (DanDeMicco) [reply src/components/ContentSidebar/ContentSidebar.js]: I named it ```isFileLoading``` to make it more clear about what is actually loading/busy. I can change it if you feel strongly that it should be changed, but I think I prefer it as is

File: src/components/ContentSidebar/ContentSidebar.js (hunk 4 above)
Comment (DanDeMicco) [reply src/components/ContentSidebar/ContentSidebar.js]: Seems good, collapses when under a certain amount of px

File: src/components/ContentSidebar/ContentSidebar.js [hunk 5]
```diff
@@ -460,6 +463,20 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+     *
+     * @private
+     * @param {Error} e - API error
+     * @return {void}
+     */
+    fetchFileErrorCallback = (e: $AxiosXHR<any>) => {
+        this.setState({
+            isFileLoading: false
+        });
+        this.errorCallback(e);
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: nit: newline

File: src/components/ContentSidebar/ContentSidebar.js [hunk 6]
```diff
@@ -1418,6 +1441,33 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+
+    /**
+     * Add classification click handler
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationClick = (): void => {
+        const { onClassificationClick = noop } = this.props.detailsSidebarProps;
+
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: Did you try setting isFileLoading here? Might be a better user experience if the user immediately sees the loading UI as a result of their action.

File: src/components/ContentSidebar/ContentSidebar.js [hunk 7]
```diff
@@ -437,6 +445,7 @@ class ContentSidebar extends PureComponent<Props, State> {
             accessStats: undefined,
             accessStatsError
         });
+
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: nice

**Pull request merged**